diff --git a/src/view/placeholder.js b/src/view/placeholder.js
index ec83640d4f7d12d8278c921a3c641917543ca859..09e96933b392f38d079ac3126ead49690e408228 100755
--- a/src/view/placeholder.js
+++ b/src/view/placeholder.js
@@ -155,16 +155,7 @@ export function needsPlaceholder( element ) {
 	const isEmptyish = !Array.from( element.getChildren() )
 		.some( element => !element.is( 'uiElement' ) );

-	// If the element is empty and the document is blurred.
-	if ( !doc.isFocused && isEmptyish ) {
-		return true;
-	}
-
-	const viewSelection = doc.selection;
-	const selectionAnchor = viewSelection.anchor;
-
-	// If document is focused and the element is empty but the selection is not anchored inside it.
-	if ( isEmptyish && selectionAnchor && selectionAnchor.parent !== element ) {
+	if ( isEmptyish ) {
 		return true;
 	}

