diff --git a/src/editorui/bodycollection.js b/src/editorui/bodycollection.js
index 28d6243667e624929adc7f04470dd9e93b408363..def417300b6ffda95e90a6550f888af7d7d9655a 100644
--- a/src/editorui/bodycollection.js
+++ b/src/editorui/bodycollection.js
@@ -93,9 +93,10 @@ export default class BodyCollection extends ViewCollection {
             children: this
         }).render();
         // Create a shared wrapper if there were none or the previous one got disconnected from DOM.
+        const documentForWrapper = document.querySelector('[name="neos-content-main"]').contentDocument;
         if (!BodyCollection._bodyWrapper || !BodyCollection._bodyWrapper.isConnected) {
-            BodyCollection._bodyWrapper = createElement(document, 'div', { class: 'ck-body-wrapper' });
-            document.body.appendChild(BodyCollection._bodyWrapper);
+            BodyCollection._bodyWrapper = createElement(documentForWrapper, 'div', { class: 'ck-body-wrapper' });
+            documentForWrapper.body.appendChild(BodyCollection._bodyWrapper);
         }
         BodyCollection._bodyWrapper.appendChild(this._bodyCollectionContainer);
     }
